docker-compose.yml

```diff
@@ -0,0 +1,11 @@
+# specify keyboard / keymap / subproject (if req) and run `docker compose up`
+# will install all deps into a docker container and build firmware, cross platform
+# Only prereq is docker
+qmk:
+  build: .
+  environment:
+    keyboard: lets_split
+    keymap: adam
+    subproject: rev2
+  volumes:
+    - .:/qmk
```

Comment:
I'm not sure if it's a good idea to keep a file that the user is supposed to edit in GIT.  I also know almost nothing about docker and docker-compose, but for me compose seems like an overkill for what we need for QMK.

I think the ideal way of using docker would be ,if we could just start docker ask it to run make with the specified paramters. The only difference from normal make would be that it runs inside the docker image.